glyphs-reader/src/plist.rs

```diff
@@ -472,6 +506,387 @@ impl From<BTreeMap<String, Plist>> for Plist {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let val: f64 = tokenizer.try_into()?;
+        Ok(Some(val))
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for Point {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let coords: Vec<f64> = parse_delimited_vec(
+            tokenizer,
+            if let Token::OpenBrace = tokenizer.peek()? {
+                CSV_IN_BRACES
+            } else {
+                CSV_IN_PARENS
+            },
+        )?;
+        if coords.len() != 2 {
+            return Err(Error::SomethingWentWrong);
+        }
+        Ok((coords[0], coords[1]).into())
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for Option<Point> {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let val: Point = tokenizer.try_into()?;
+        Ok(Some(val))
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for i64 {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        match tokenizer.lex()? {
+            Token::Atom(val) => parse_int(val),
+            Token::String(val) => parse_int(&val),
+            _ => Err(Error::ExpectedNumber),
+        }
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for Option<i64> {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let val: i64 = tokenizer.try_into()?;
+        Ok(Some(val))
+    }
+}
+
+impl TryFrom<&crate::plist::Tokenizer<'_>> for Vec<i64> {
+    type Error = crate::plist::Error;
+    fn try_from(tokenizer: &crate::plist::Tokenizer<'_>) -> Result<Self, Self::Error> {
+        parse_vec(tokenizer)
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for bool {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        match tokenizer.lex()? {
+            Token::Atom(val) => parse_int(val).map(|v| v == 1),
+            _ => Err(Error::ExpectedNumber),
+        }
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for Option<bool> {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let val: bool = tokenizer.try_into()?;
+        Ok(Some(val))
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for String {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        match tokenizer.lex()? {
+            Token::Atom(val) => Ok(val.to_string()),
+            Token::String(val) => Ok(val.to_string()),
+            _ => Err(Error::ExpectedString),
+        }
+    }
+}
+
+impl TryFrom<&Tokenizer<'_>> for Option<String> {
+    type Error = Error;
+
+    fn try_from(tokenizer: &Tokenizer<'_>) -> Result<Self, Self::Error> {
+        let val: String = tokenizer.try_into()?;
+        Ok(Some(val))
+    }
+}
+
+impl TryFrom<&crate::plist::Tokenizer<'_>> for Vec<String> {
```

Comment:
As you've noted we should definitely have blanket impls for `Vec` and `Option`, and this should definitely be possible.

Reply:
I agree but my on-plane attempt did not compile. Let me tidy up per your other comments nad then I may ping you to help figure out why not if it still won't cooperate.

Reply:
After cleanup blanket impl is fine. I don't entirely understand why it wasn't before but I don't have infinite time to pursue that.